### PR TITLE
Flow

### PR DESCRIPTION
**Pull request by flexjames**
Associated Issue: #1785

Added Flow to client.js & dehydrate-state.js.

### CODE REVIEW

File: src/utils/client.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
+// @flow
+
 const { firefox } = require("devtools-client-adapters");
 const { prefs } = require("./prefs");
 
-async function onFirefoxConnect(actions) {
+async function onFirefoxConnect(actions: Object) {
```
Comment (wldcordeiro) [inline src/utils/client.js]: I'm not sure if typing these as `Object` really buys us much, could we use a proper type that maps to the actions the client receives @jasonLaster

File: src/utils/dehydrate-state.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
+// @flow
+
 const I = require("immutable");
 const { fromJS } = I;
-const SourcesState = require("../reducers/sources").State;
+const SourcesState: any = require("../reducers/sources").State;
```
Comment (jasonLaster) [inline src/utils/dehydrate-state.js]: we can use `SourcesState` that's exported in the reducer

File: src/utils/client.js [hunk 3]
```diff
@@ -33,7 +35,7 @@ async function onFirefoxConnect(actions) {
   }
 }
 
-async function onConnect(connection, actions) {
+async function onConnect(connection: Object, actions: Object) {
```
Comment (jasonLaster) [inline src/utils/client.js]: hmm, we can add a local type:

```js
import type { Tab } from "../types"
type Connection = { tab: Tab }
```

File: src/utils/client.js (hunk 1 above)
Comment (jasonLaster) [inline src/utils/client.js]: true,, we can import Actions

```js
import type { Actions } from "../actions/types";
```

File: src/utils/client.js (hunk 3 above)
Comment (jasonLaster) [inline src/utils/client.js]: Hmm, this isn't used here. so we can ignore it for the time being.

Comment (jasonLaster) [inline src/utils/client.js]: Also, we don't need to type `actions` because it's the object that holds all of the actions, which is currently untyped

**Pull request merged**